Simplified/Reader2/Networking/NYPLLastReadPositionSynchronizer.swift

```diff
@@ -25,21 +25,16 @@ class NYPLLastReadPositionSynchronizer: NYPLLastReadPositionSynchronizing {
     case stay, moveToServerLocator
   }
 
-  private let failFastNetworkExecutor: NYPLNetworkExecutor
-  let synchronizer: NYPLAnnotationSyncing.Type
+  let synchronizer: NYPLLastReadPositionSupportAPI
 
   /// Designated initializer.
   ///
   /// - Parameters:
   ///   - bookRegistry: The registry that stores the reading progresses.
   init(bookRegistry: NYPLBookRegistryProvider,
-       synchronizer: NYPLAnnotationSyncing.Type) {
+       synchronizer: NYPLLastReadPositionSupportAPI) {
     self.bookRegistry = bookRegistry
     self.synchronizer = synchronizer
-    failFastNetworkExecutor = NYPLNetworkExecutor(
```

Comment:
this was moved inside NYPLAnnotations because it belongs there with the networking logic. Ideally we should dependency-inject it into the NYPLAnnotations initializer.